File: android/src/main/java/expo/modules/richinput/ExpoRichInputModule.kt
package expo.modules.richinput

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoRichInputModule : Module() {
    override fun definition() = ModuleDefinition {

        Name("ExpoRichInput")

        View(RichInputView::class) {

            Events(
                "onEditEvent",
                "onKeyboardAction",
                "onSelectionChange"
            )

            Prop("editable") {
                view: RichInputView, value: Boolean ->
                view.isEnabled = value
            }

            AsyncFunction("focus") {
                view: RichInputView ->
                view.focusInput()
            }

            AsyncFunction("blur") {
                view: RichInputView ->
                view.blurInput()
            }
        }
    }
}

File: ios/ExpoRichInputModule.swift
import ExpoModulesCore

public class RichInputModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoRichInput")

        View(RichInputView.self) {

            // MARK: Events exposed to JS
            Events("onEditEvent", "onKeyboardAction")

            // MARK: Functions callable from JS
            // ref.current.focus()
            AsyncFunction("focus") {
                (view: RichInputView) in
                DispatchQueue.main.async {
                    view.focusInput()
                }
            }

            // ref.current.blur()
            AsyncFunction("blur") {
                (view: RichInputView) in
                DispatchQueue.main.async {
                    view.blurInput()
                }
            }
        }
    }
}